presto-spark/src/main/java/com/facebook/presto/spark/planner/SparkRddPlanner.java

```diff
@@ -0,0 +1,254 @@
+            }
+
+            List<SubPlanWithTaskSources> children = subPlan.getChildren();
+            checkArgument(
+                    remoteSources.size() == children.size(),
+                    "number of remote sources doesn't match the number of child stages: %s != %s",
+                    remoteSources.size(),
+                    children.size());
+
+            if (children.size() == 1) {
+                // Single remote source
+                SubPlanWithTaskSources childSubPlan = getOnlyElement(children);
+                JavaPairRDD<Integer, byte[]> childRdd = createRdd(childSubPlan);
+
+                PlanFragment childFragment = childSubPlan.getFragment();
+                RemoteSourceNode remoteSource = getOnlyElement(remoteSources);
+                List<PlanFragmentId> sourceFragmentIds = remoteSource.getSourceFragmentIds();
+                checkArgument(sourceFragmentIds.size() == 1, "expected to have exactly only a single source fragment");
+                checkArgument(childFragment.getId().equals(getOnlyElement(sourceFragmentIds)));
+
+                SparkTaskDescriptor sparkTaskDescriptor = createTaskDescriptor(fragment, ImmutableList.of());
+                byte[] serializedTaskDescriptor = sparkTaskRequestJsonCodec.toJsonBytes(sparkTaskDescriptor);
+
+                PartitioningHandle partitioning = fragment.getPartitioning();
+                if (partitioning.equals(COORDINATOR_DISTRIBUTION)) {
+                    // TODO: We assume COORDINATOR_DISTRIBUTION always means OutputNode
+                    // But it could also be TableFinishNode, in that case we should do collect and table commit on coordinator.
+
+                    // TODO: Do we want to return an RDD for root stage? -- or we should consider the result will not be large?
+                    List<Tuple2<Integer, byte[]>> collect = childRdd.collect();
+                    List<Tuple2<Integer, byte[]>> result = ImmutableList.copyOf(compilerFactory.create().compile(
+                            0,
+                            serializedTaskDescriptor,
+                            ImmutableMap.of(remoteSource.getId().toString(), collect.iterator()),
+                            taskStatsCollector));
+                    return sparkContext.parallelizePairs(result);
+                }
+                else if (partitioning.equals(FIXED_HASH_DISTRIBUTION) ||
+                        // when single distribution - there will be a single partition 0
+                        partitioning.equals(SINGLE_DISTRIBUTION)) {
+                    String planNodeId = remoteSource.getId().toString();
+                    return childRdd
+                            // TODO: What's the difference of using
+                            //  partitionBy/mapPartitionsToPair vs. groupBy vs. mapToPair ???
+                            .partitionBy(partitioning.equals(FIXED_HASH_DISTRIBUTION) ? new HashPartitioner(hashPartitionCount) : new HashPartitioner(1))
```

Comment:
>> We can implement our own IntegerIdentityPartitioner

@wenleix : Maybe I missed something obvious. Why are you proposing to implement to custom partitioner ?

Reply:
Also, there is already a `PartitionIdPassthrough` which is kinda an identity partitioner

Reply:
@tejasapatil I cannot find it in the classpath. Is it part of the spark-sql package?

Reply:
@tejasapatil: When executing Presto-on-Spark, Presto's `PartitionedOutputoOperator` already calculate the output partition for each row (or row batch). 

Thank you so for pointing out `PartitionIdPassthrough`! That's exactly what we need, but it's a private class in SparkSQL. Although this sort of suggests `SparkSQL` requires similar functionality :) https://github.com/apache/spark/blob/b389b8c5f0650a7e63098f18437fcaa29998732a/sql/core/src/main/scala/org/apache/spark/sql/execution/ShuffledRowRDD.scala#L41-L47


On a side note, in a long term it might be interesting to create specific RDD/Partition class related to `PrestoSpark` to improve the code readability, similar to what did in `SparkSQL`. cc @sameeragarwal , @arhimondr